### PR TITLE
tip for mixed dtypes

### LINKED ISSUE #4035: Add tip in regarding mixed dtypes to profiling output.
See @lamblin's comment in #3964.

Context : the logistic regression example from the deep learning tutorials was used to provide an example of the output of Theano's profiling mode. However, in the profile output, it seems like one of the dot products has been kept in the graph because of float32/float64 dtypes (otherwise it could have been removed).

### PR DESCRIPTION
**Pull request by AnishShah**
Fixes #4035

### CODE REVIEW

File: theano/compile/profiling.py [hunk 1]
```diff
@@ -1391,6 +1391,16 @@ def exp_float32_op(op):
                           " generator supported on the GPU.", file = file)
                 break
 
+        # tip 6
+        for a in self.apply_time:
+            node = a
+            if (isinstance(node.op, T.Dot) and
+                len(set(i.dtype for i in node.inputs)) == 1):
```
Comment (nouiz) [inline theano/compile/profiling.py]: I think it should be == 2.

Comment (nouiz) [inline theano/compile/profiling.py]: Or better, != 1

**Comment by AnishShah** [comment]
@nouiz I have updated the PR

File: theano/compile/profiling.py [hunk 2]
```diff
@@ -1391,6 +1391,18 @@ def exp_float32_op(op):
                           " generator supported on the GPU.", file = file)
                 break
 
+        # tip 6
+        for a in self.apply_time:
+            node = a
+            print("dtype", [i.dtype for i in node.inputs])
+            print("dim", [i.ndim for i in node.inputsg])
```
Comment (nouiz) [inline theano/compile/profiling.py]: inputsg -> input

This is what make travis fail. In fact, remove those 2 print. We dont, want them printed.

Comment (AnishShah) [inline theano/compile/profiling.py]: Oops. Very sorry. I was trying to run the profiling example

**Comment by AnishShah** [comment]
@nouiz Updated the PR

**Comment by nouiz** [comment]
thanks

**Pull request merged**